device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1418,7 +1418,7 @@ private void handleMessageException(IotHubTransportPacket packet, TransportExcep
                 RetryDecision retryDecision = config.getRetryPolicy().getRetryDecision(packet.getCurrentRetryAttempt(), transportException);
                 if (retryDecision.shouldRetry())
                 {
-                    this.taskScheduler.schedule(new MessageRetryRunnable(this.waitingPacketsQueue, packet, this), retryDecision.getDuration(), MILLISECONDS);
```

Comment:
Using ```this``` as the lock parameter led to the IotHubTransport instance being notified when the message to retry had be requeued rather than notifying the ```sendThreadLock``` that we use for keeping the worker thread active/asleep

Reply:
This previous parameter was just a mistake, not by design